### PR TITLE
Add support for attach_metadata into PodMonitor

### LINKED ISSUE #4779: Add support for `attach_metadata` into ServiceMonitor/PodMonitor
**What is missing?**

As of Prometheus 2.35 we can add the `attach_metadata` option to a `kubernetes_sd_config` to allow us to add `node` labels to pods discovered in the scrape config.

**Why do we need it?**

To be able to expose node metadata as a label on pod metrics, such as for `node_exporter` metrics. This is useful for monitoring items such as `nf_conntrack` on hosts and looking up the node by hostname rather than instance IP.

**Environment**

* Prometheus Operator version: `0.56.2`

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by mikechengwei**
## Description

Close #4779 .

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Add support for attach_metadata into PodMonitor
```

### CODE REVIEW

File: pkg/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -1189,6 +1189,14 @@ type PodMonitorSpec struct {
 	// Per-scrape limit on length of labels value that will be accepted for a sample.
 	// Only valid in Prometheus versions 2.27.0 and newer.
 	LabelValueLengthLimit uint64 `json:"labelValueLengthLimit,omitempty"`
+	//Attaches node metadata to discovered targets. Only valid for role: pod.
+	//Only valid in Prometheus versions 2.35.0 and newer.
```
Comment (fpetkovski) [inline pkg/apis/monitoring/v1/types.go:1193]:
```suggestion
	// Attaches node metadata to discovered targets. Only valid for role: pod.
	// Only valid in Prometheus versions 2.35.0 and newer.
```

File: pkg/prometheus/promcfg.go [hunk 2]
```diff
@@ -1547,6 +1548,13 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 		// config as well, make sure to path the right namespace here.
 		k8sSDConfig = addTLStoYaml(k8sSDConfig, "", apiserverConfig.TLSConfig)
 	}
+	if attachMetadata != nil {
```
Comment (fpetkovski) [inline pkg/prometheus/promcfg.go:1551]: We should make sure to attach this key only when the version is greater or equal 2.35.0. You can see an example of this pattern here: https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go#L192

**Review by fpetkovski — COMMENTED** [review]
Thanks for the contribution. Overall this looks good with just a few small nits

File: pkg/prometheus/promcfg.go (hunk 2 above)
Comment (mikechengwei) [reply pkg/prometheus/promcfg.go:1551]: ok.

**Comment by mikechengwei** [comment]
@fpetkovski plz help me review it again.

**Review by fpetkovski — APPROVED** [review]
This looks good to me now 👍

**Review by simonpasquier — APPROVED** [review]
thanks!

**Pull request merged**